### PR TITLE
Fix hledger-lib build with ghc 8.0.x and base-compat 0.10.x.

### PR DESCRIPTION
**Pull request by peti**
We don't need to import Data.Monoid because Prelude.Compat exports "<>"
already. In fact, importing that module causes build failures:

    Hledger/Read/Common.hs:725:62: error:
        Ambiguous occurrence ‘<>’
        It could refer to either ‘Sem.<>’,
                                 imported from ‘Prelude.Compat’ at Hledger/Read/Common.hs:97:1-39
                                 (and originally defined in ‘Data.Semigroup’)
                              or ‘Data.Monoid.<>’,
                                 imported from ‘Data.Monoid’ at Hledger/Read/Common.hs:110:1-18

Fixes https://github.com/simonmichael/hledger/issues/794.

### CODE REVIEW

**Comment by simonmichael** [comment]
We should probably update all four base-compat bounds to >=0.10.1 && <0.11 too I guess.

**Comment by peti** [comment]
OK, done.

**Comment by simonmichael** [comment]
It looks like we can't use base-compat 0.10.* yet with hledger-api, since servant deps don't allow it. I guess you can add the older one in stack.yaml extra-deps ? Also add the newer one there, for folks building with cabal or older stack snapshots.

Also ==0.10.* isn't quite the same thing, wouldn't it be better to depend on >=0.10.1 ?

**Comment by simonmichael** [comment]
Ie in stack.yaml:
```
extra-deps:
  ...
  - base-compat-0.10.1
  # for hledger-api:
  - base-compat-0.9.3
```

**Comment by simonmichael** [comment]
..and in the package.yamls:
```
- base-compat >=0.10.1 && <0.11
```

**Comment by simonmichael** [comment]
Phew, details. Let me try again, shouldn't it be maybe like this:

stack.yaml: add the old base-compat to extra-deps
```
  # for hledger-api:
  - base-compat-0.9.3
```
stack-ghc*.yaml: add the new base-compat to extra-deps
```
  - base-compat-0.10.1
```
hledger{,-lib,-ui,-web}: require the latest base-compat
```
  - base-compat >=0.10.1 && <0.11
```

**Comment by simonmichael** [comment]
With stack.yaml, it's also failing on:
```
Test suite failure for package hledger-lib-1.9.99
    doctests:  exited with: ExitFailure 1
Full log available at /home/simon/src/hledger.org/.stack-work/logs/hledger-lib-1.9.99-test.log



    Hledger/Data/StringFormat.hs:17:1: error:
        Ambiguous module name ‘Prelude.Compat’:
          it was found in multiple packages:
          base-compat-0.10.1 base-compat-batteries-0.10.1
       |
    17 | import Prelude.Compat
       | ^^^^^^^^^^^^^^^^^^^^^
```

**Comment by peti** [comment]
> Also ==0.10.* isn't quite the same thing, wouldn't it be better to depend on >=0.10.1 ?

Why? AFAIK, base-compat 0.10.0 works fine, so why would you like to exclude it?

**Comment by peti** [comment]
https://hackage.haskell.org/package/servant-server-0.13.0.1 adds support for `base-compat-0.10.x`. Maybe that simplifies matters?

**Comment by peti** [comment]
I pushed an updated version. Running test builds now.
